### PR TITLE
AP_RangeFinder: add prearm for analogue/pwm pin conflicts

### PR DESCRIPTION
**Pull request by peterbarker**
Swiped from the RPM equivalent.

image

Tested on CubeOrange

### CODE REVIEW

File: libraries/AP_RangeFinder/AP_RangeFinder_Backend.h [hunk 1]
```diff
@@ -69,6 +69,8 @@ class AP_RangeFinder_Backend
     // quality is not available
     virtual bool get_signal_quality_pct(uint8_t &quality_pct) const { return false; }
 
+    RangeFinder::Type allocated_type() const { return _backend_type; }
```
Comment (rmackay9) [inline libraries/AP_RangeFinder/AP_RangeFinder_Backend.h:72]: maybe put a comment above this method just to save the ready 2 sec in figuring it out?

File: libraries/AP_RangeFinder/AP_RangeFinder.cpp [hunk 2]
```diff
@@ -838,6 +838,32 @@ bool RangeFinder::prearm_healthy(char *failure_msg, const uint8_t failure_msg_le
...
+        switch (drivers[i]->allocated_type()) {
+        case Type::ANALOG:
+        case Type::PX4_PWM:
+        case Type::PWM: {
+            // ensure pin is configured
+            if (params[i].pin == -1) {
+                hal.util->snprintf(failure_msg, failure_msg_len, "RNGFND%u_PIN not set", unsigned(i + 1));
+                return false;
+            }
+            // ensure that the pin we're configured to use is available:
```
[the remark this answers is not in the document]
Comment (peterbarker) [reply libraries/AP_RangeFinder/AP_RangeFinder.cpp]: The continuation of that comment is the code following it.  I like having the colon there as it says to the reader, "more to come" or "read on".  Kind of like "don't do silly things: silly things are bad for your health" (not that that's particularly good grammar, but whatever....)

I've removed it :-)

**Review by andyp1per — APPROVED** [review]

**Comment by rmackay9** [comment]
No big deal but we somehow missed adding this to the Copter-4.2 project (it was only included in the Plane-4.2 project).  I've added it now.

**Comment by rmackay9** [comment]
I've included this in Copter/Rover-4.2.3-rc1 but it is not included in Plane-4.2.3.  It looks useful and low risk to me though.

**Pull request merged**